.github/workflows/se-test.yml

```diff
@@ -11,17 +11,11 @@ jobs:
     - uses: actions/checkout@v3
     - uses: actions/setup-python@v4
       with:
-        python-version: '3.x-dev'
```

Comment:
Why did we remove -dev from here, but in a ad722fcd399811202f7f94e79dca3acf1e2bab8b we specified 3.x-dev?

Reply:
I tried `3.x-dev` as https://github.com/actions/setup-python#available-versions-of-python says that it'd get the latest patch version thanks to `dev`. It didn't exist in the manifest when I tried running the action locally with `act` though:
```
::error::Version 3.x-dev with arch x64 not found%0AThe list of all available versions can be found here: https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json
```

`3.x` seems to be the most common version I see for Github Actions, e.g. in the first basic snippet here https://github.com/actions/setup-python#usage.

One strange thing: https://github.com/actions/setup-python/releases/tag/v4.0.0 has a commit to fix the `3.x-dev` syntax. I think the bug is unrelated to not parsing the version and finding best patch version in the manifest though 🤔

Reply:
`3.11-dev` works
3.x must be using latest minor (and patch, implicitly) hence `.x-dev` is ignored